### PR TITLE
Repo structure and k8s client

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1086

- Repo structure with readme, license etc
- Small initial scope with util for creating Kubernetes Clients

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - error
+  - logger
+- package: k8s.io/client-go
+  version: v2.0.0
+  subpackages:
+  - kubernetes
+  - rest
+testImport:
+- package: github.com/stretchr/testify
+  version: ^1.1.4
```
Comment (JosephSalisbury) [inline glide.yaml]: Random drive by, let's pin this

File: client/k8s/k8s.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	var err error
+
+	if config.InCluster {
+		config.Logger.Log("debug", "creating in-cluster config")
+		rawClientConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, microerror.MaskAny(err)
+		}
+
+		if config.Address != "" {
```
Comment (kopiczko) [inline client/k8s/k8s.go]: Does it make sense to use external API for in-cluster operator?

Comment (rossf7) [inline client/k8s/k8s.go]: Good point, I've removed.

File: client/k8s/k8s.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+func getRawClientConfig(config Config) (*rest.Config, error) {
+	var rawClientConfig *rest.Config
+	var err error
+
+	if config.InCluster {
+		config.Logger.Log("debug", "creating in-cluster config")
+		rawClientConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, microerror.MaskAny(err)
+		}
```
Comment (kopiczko) [inline client/k8s/k8s.go]: Maybe a nitpicking, but may be replaced with `return rawClientConfig, microerror.MaskAny(err)`. Then `else` is not needed.

Comment (kopiczko) [inline client/k8s/k8s.go]: Also `rawClientConfig` and `err` won't need to be declared on top.

**Comment by rossf7** [comment]
@xh3b4sd coool. There is a WIP PR in cert-operator showing it being used. It needs a rebase as its pretty old but I'll get that sorted.

https://github.com/giantswarm/cert-operator/pull/28/files

**Comment by kopiczko** [comment]
wip

**Pull request merged**